test/powershell/Modules/Microsoft.PowerShell.Utility/WebCmdlets.Tests.ps1

```diff
@@ -143,6 +142,125 @@ function GetTestData
     return $body
 }
 
+<#
+    Extract the raw content from a response and
+    convert the embedded json to PSObjects.
+    Assumes the raw content is from the  HTTPListener.
+#>
+function Convert-ResponseRawContent
+{
+    param
+    (
+        [Parameter(Mandatory)]
+        [ValidateNotNull()]
+        [Microsoft.PowerShell.Commands.WebResponseObject] $Response
+    )
+
+    $index = $Response.RawContent.IndexOf("{")
+    [string] $rawContent = $Response.RawContent.SubString($index)
+    return ($rawContent | ConvertFrom-Json)
+}
+
+function ExecuteRedirectRequest
+{
+    param (
+        [Parameter(Mandatory)]
+        [string]
+        $uri,
+
+        [switch] $UsePost,
+
+        [switch] $PreserveAuthorizationOnRedirect
+    )
+    $result = [PSObject]@{Output = $null; Error = $null; Content = $null}
+
+    try
+    {
+        $headers = @{"Authorization" = "test"}
+        if ($UsePost)
+        {
+            $result.Output = Invoke-WebRequest -Uri $uri -TimeoutSec 5 -Headers $headers -PreserveAuthorizationOnRedirect:$PreserveAuthorizationOnRedirect.IsPresent -Method POST
+        }
+        else
+        {
+            $result.Output = Invoke-WebRequest -Uri $uri -TimeoutSec 5 -Headers $headers -PreserveAuthorizationOnRedirect:$PreserveAuthorizationOnRedirect.IsPresent
+        }
+
+        $result.Content = Convert-ResponseRawContent -Response $result.Output
+    }
+    catch
+    {
+        $result.Error = $_
+    }
+
+    return $result
+}
+
+Describe "Invoke-WebRequest redirect tests" -Tags ("Feature", "Redirect") {
+
+    BeforeAll {
+        $null = Start-HttpListener -AsJob -Port 8082
+    }
+
+    AfterAll {
+        $null = Stop-HttpListener -Port 8082
+    }
+
+    It "Validates Invoke-WebRequest with -PreserveAuthorizationOnRedirect preserves the authorization header 302 (Found) redirect" {
+        $response = ExecuteRedirectRequest -Uri 'http://localhost:8082/PowerShell?test=redirect&type=Found' -PreserveAuthorizationOnRedirect
+        $response.Error | Should BeNullOrEmpty
+
+        # ensure Authorization header has been preserved.
+        $response.Content.Headers -contains "Authorization" | Should Be $true
+    }
+
+    It "Validates Invoke-WebRequest preserves the authorization header on multiple redirects." {
+        $response = ExecuteRedirectRequest -Uri "http://localhost:8082/PowerShell?test=redirect&type=Found&multiredirect=true" -PreserveAuthorizationOnRedirect
+
+        $response.Error | Should BeNullOrEmpty
+        # ensure Authorization header was stripped
+        $response.Content.Headers -contains "Authorization" | Should Be $true
```

Comment:
Actually, I can't fix it without fixing ConvertTo-Json.   $request.Headers is a WebHeaderCollection which is a NamedValueCollection (ICollection).  ConvertTo-Json (and Select-Object for that matter) treat it as a collection and would have to special case the type.